File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int maxi = 0;
        int cur_min = prices[0];

        for (int i = 1;i<prices.size();i++){
            maxi = max(maxi, prices[i] - cur_min);
            cur_min = min(cur_min, prices[i]);

        }

        return maxi;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini = prices[0]
        max_profit = 0
         
        for i in range(1,len(prices)):
            if (prices[i] > mini):
                max_profit = max(max_profit, prices[i]-mini)
            else:
                mini = min(mini, prices[i])

        return max_profit 